src/integrationtests/java/com/aws/greengrass/integrationtests/provisioning/ProvisionFromTestPlugin.java

```diff
@@ -0,0 +1,267 @@
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_AWS_REGION;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_CERTIFICATE_FILE_PATH;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_IOT_DATA_ENDPOINT;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_PRIVATE_KEY_PATH;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_ROOT_CA_PATH;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_THING_NAME;
+import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionUltimateCauseOfType;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@ExtendWith(GGExtension.class)
+public class ProvisionFromTestPlugin extends BaseITCase {
+
+    private Kernel kernel;
+
+    @TempDir
+    Path configurationFilePath;
+    @TempDir
+    Path certFilePath;
+    @TempDir
+    Path privateKeyPath;
+    @TempDir
+    Path rootCAPath;
+
+    @BeforeEach
+    void before() {
+        kernel = new Kernel();
+    }
+
+    @AfterEach
+    void after() {
+        if (kernel != null) {
+            // Executor service is not able to terminate threads in some tests within the default 30 seconds
+            kernel.shutdown(40);
+        }
+    }
+
+    @Order(1)
+    @Test
+    @SuppressWarnings("PMD.AvoidDeeplyNestedIfStmts")
+    void GIVEN_nucleus_started_with_testProvisioningPlugin_AND_without_config_params_THEN_device_runs_offline_mode() throws Throwable {
+        URL filepath = getClass().getResource("config_without_test_provisioning_plugin.yaml");
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel, filepath);
+        CountDownLatch logLatch =  new CountDownLatch(2);
+        CountDownLatch reverseLatch = new CountDownLatch(1);
+        DeviceConfiguration deviceConfiguration = kernel.getContext().get(DeviceConfiguration.class);
+        try (AutoCloseable listener = TestUtils.createCloseableLogListener((message) -> {
+            String messageString = message.getMessage();
+            if (messageString.contains("Device not configured to talk to AWS Iot cloud")) {
+                logLatch.countDown();
+            } else if ("Updated provisioning configuration".equals(messageString)) {
```

Comment:
Nit - Looks like these logs you're verifying here and other tests are from Nucleus launch code, could add constants for these and reference them here so you don't need to update tests if these logs change